lib/compiler.js

```diff
@@ -637,9 +659,25 @@ function buildAllMixinInstructions(appRootDir, mixinDirs, options) {
     return meta;
   });
 
+  if (includeMixins.length > 0) {
+    mixins = mixins.filter(function(m) {
+      return (includeMixins.indexOf(m.name) === -1 ? false : true);
+    });
+  }
+
   return mixins;
 }
 
+function fetchMixinsUsedInModels(modelInstructions) {
+  var modelMixins = [];
+  modelInstructions.forEach(function(model) {
+    if (model.definition && model.definition.mixins) {
+      modelMixins = modelMixins.concat(Object.keys(model.definition.mixins));
+    }
+  });
+  return modelMixins;
```

Comment:
You can use [_.flatten](https://lodash.com/docs#flatten) to make the filtering code simpler.

```js
return _(modelInstructions)
  .map(function(model) {
    return model.definition && model.definition.mixins ? Object.keys(model.definition.mixins) : [];
  })
  .flatten();
```